### PR TITLE
Windows: linker compatibility with AppContainer for winuwp target

### PR DESCRIPTION
**Pull request by clarkezone**
This PR is the final engine patch required for the Flutter Engine `--winuwp` target to build and link correctly for the UWP/AppContainer environment.  Whist this PR stands alone, the linker configuration will not be completely compatible until the following related changes also land:

- [Add UWP support (I4ed9e440) · Gerrit Code Review (googlesource.com)](https://skia-review.googlesource.com/c/skia/+/345219)
- [Add support for building in Flutter Windows UWP configuration (Idf5a4cbb) · Gerrit Code Review (googlesource.com)](https://chromium-review.googlesource.com/c/angle/angle/+/2620580) 
- https://github.com/flutter/buildroot/pull/438

Related issue:
https://github.com/flutter/flutter/issues/70196
https://github.com/flutter/flutter/issues/14967

Must land before:
https://github.com/flutter/buildroot/pull/438

Related:

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: shell/platform/windows/key_event_handler.h [hunk 1]
```diff
@@ -27,8 +27,13 @@ class KeyEventHandler : public KeyboardHookHandler {
   using SendInputDelegate =
       std::function<UINT(UINT cInputs, LPINPUT pInputs, int cbSize)>;
 
+#ifdef WINUWP
+  explicit KeyEventHandler(flutter::BinaryMessenger* messenger,
```
Comment (cbracken) [inline shell/platform/windows/key_event_handler.h:31]: Add an explanatory comment/todo here.

File: shell/platform/windows/key_event_handler.cc [hunk 2]
```diff
@@ -190,6 +192,7 @@ void KeyEventHandler::HandleResponse(bool handled,
       std::cerr << "Unable to find event " << id << " in pending events queue.";
       return;
     }
+#ifndef WINUWP
```
Comment (cbracken) [inline shell/platform/windows/key_event_handler.cc:195]: Add an explanatory comment/todo here referring to the null delegate.

File: shell/platform/windows/key_event_handler.cc [hunk 3]
```diff
@@ -122,7 +122,9 @@ KeyEventHandler::KeyEventHandler(flutter::BinaryMessenger* messenger,
               kChannelName,
               &flutter::JsonMessageCodec::GetInstance())),
       send_input_(send_input) {
+#ifndef WINUWP
```
Comment (cbracken) [inline shell/platform/windows/key_event_handler.cc:125]: Add an explanatory comment/todo here referring to the null delegate.

**Review by cbracken — APPROVED** [review]
lgtm with minor nits.

File: shell/platform/windows/key_event_handler.h (hunk 1 above)
Comment (clarkezone) [reply shell/platform/windows/key_event_handler.h:31]: Fixed in next commit

File: shell/platform/windows/key_event_handler.cc (hunk 2 above)
Comment (clarkezone) [reply shell/platform/windows/key_event_handler.cc:206]: Fixed in next commit.

File: shell/platform/windows/key_event_handler.cc (hunk 3 above)
Comment (clarkezone) [reply shell/platform/windows/key_event_handler.cc:128]: Fixed in next commit.

**Pull request merged**